### ISSUE TITLE
[bug] when dim=1

### ISSUE THREAD

**Issue by Gabibing** [lucidrains/vector-quantize-pytorch#228]
## issue
When using ResidualVQ with feature dimension dim=1 and passing a 2D mask of shape [B, T], masking is not handled correctly. 
When mask=None is used as a workaround, the output’s batch dimension collapse to 1.
```
vq = ResidualVQ(dim=1, num_quantizers=1, codebook_size=4)
```

## error log
```
vector_quantize_pytorch.py", line 1349, in forward
[rank0]:     quantize = einx.where(
                        ^^^^^^^^^^^
einx.expr.stage3.SolveValueException: Failed to solve values of expressions.
Found contradictory values {5936, 1} for equivalent expressions {'d', '5936', '1'}
Input:
    'b n = 8 742'
    'b n d = 8 742 5936'
    'b n d = 8 742 1'
    'b n d = None'
```

**Comment by lucidrains** [comment]
@Gabibing can you make sure it is a runnable script and produces the error?

**Comment by Gabibing** [comment]
@lucidrains 
## code
```
import torch
import torch.nn as nn
import torch.nn.functional as F


from vector_quantize_pytorch import ResidualVQ

class ContentQuantizer(nn.Module):
    def __init__(self, in_dim=1024, hidden_dim=256, codebook_size=2048):
        super().__init__()
        self.net = nn.Linear(in_dim, hidden_dim)
        self.quantizer = ResidualVQ(
            dim=hidden_dim,
            num_quantizers=1,
            codebook_size=codebook_size,
            stochastic_sample_codes=True,
            sample_codebook_temp=1.0,
        )

    def forward(self, x, mask, sample_codebook_temp=0.):  # x: [B,T,1024]
        x = self.net(x)    # [B,T,hidden_dim]
        print(x.shape)
        quantized, indices, commit_loss = self.quantizer(x, mask=mask, sample_codebook_temp=sample_codebook_temp)
        return quantized, indices, commit_loss

if __name__ == '__main__':
   vq = ContentQuantizer(1024, 1, 2048)

   x = torch.randn(8, 700, 1024)
   mask = torch.ones(8, 700, dtype=torch.bool)
   q, i, c = vq(x, mask)
   print(q.shape)
```

## Error
```
einx.expr.stage3.SolveValueException: Failed to solve values of expressions. Found contradictory values {5600, 1} for equivalent expressions {'d', '5600', '1'}
Input:
    'b n = 8 700'
    'b n d = 8 700 5600'
    'b n d = 8 700 1'
    'b n d = None'
```

**Comment by Gabibing** [comment]
@lucidrains It works perfectly. Thank you!
